engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/typeinformation/TypeInformationDetection.scala

```diff
@@ -20,3 +20,12 @@ trait TypeInformationDetection extends Serializable {
   def forValueWithContext[T](validationContext: ValidationContext, value: TypingResult): TypeInformation[ValueWithContext[T]]
 
 }
+
+/**
+  * Trait that allows for providing more details TypeInformation when TypingResult is known.
+  */
+trait TypeInformationDetectionForTypingResult extends TypeInformationDetection {
```

Comment:
1. Why not to merge it with TypeInformationDetection? 2. Maybe it could have better name because now it looks like it only detect type for typing result (fot context and so on - no)

Reply:
:+1: If it extends TypeInformationDetection I think it's better to move this method to forType?

Reply:
Fixed. Removed TypeInformationDetectionForTypingResult, method forType moved to TypeInformationDetection.